riscv/sim.cc

```diff
@@ -45,7 +44,7 @@ sim_t::sim_t(const cfg_t *cfg, const char* varch, bool halted, bool real_time_cl
     cfg(cfg),
     mems(mems),
     plugin_devices(plugin_devices),
-    procs(std::max(cfg->nprocs(), size_t(1))),
+    procs(std::max(cfg->hartids().size(), size_t(1))),
```

Comment:
Perhaps there should be a method `cfg_t::nprocs()` that returns `hartids().size()`?

Reply:
Good idea. Done.